s3/replication/manager/tests/system/load_transfer.py

```diff
@@ -0,0 +1,294 @@
+
+# Helper methods
+
+
+class ObjectDataGenerator:
+    def __init__(self, logger, object_name, object_size):
+        """Initialise."""
+        self._logger = logger
+        self._object_name = object_name
+        self._object_size = object_size
+        self._hash = hashlib.md5()
+        self._state = S3RequestState.INITIALISED
+
+    def get_state(self):
+        """Returns current request state."""
+        return self._state
+
+    def get_md5(self):
+        if self._state == S3RequestState.COMPLETED:
+            return self._md5
+        return None
+
+    async def fetch(self, chunk_size):
+        pending_size = self._object_size
+        # Generate only one chunk and just keep sending same for each iteration
+        data = os.urandom(chunk_size)
+        self._state = S3RequestState.RUNNING
+        self._logger.debug("Start data gen. pending_size {} bytes.".
+                           format(pending_size))
+        while pending_size > 0:
+            if pending_size >= chunk_size:
+                pending_size -= chunk_size
+                self._hash.update(data)
+                self._logger.debug("pending_size {} bytes.".
+                                   format(pending_size))
+                self._logger.debug("Yielding data of size {} bytes.".
+                                   format(chunk_size))
+                yield data
+            else:
+                # pending size is less than chunk size, generate only pending
+                # size
+                last_data = os.urandom(pending_size)
+                self._hash.update(last_data)
+                self._logger.debug(
+                    "Yielding last unit of data of size {} bytes.".
+                    format(pending_size))
+                yield last_data
+                pending_size = 0
+        self._logger.debug("Data generation completed for object {}.".
+                           format(self._object_name))
+        self._md5 = self._hash.hexdigest()
+        self._state = S3RequestState.COMPLETED
+
+
+def init_logger():
+    log_config_file = os.path.join(os.path.dirname(__file__),
+                                   'config', 'logger_config.yaml')
+
+    print("Using log config {}".format(log_config_file))
+    logger = setup_logger('client_tests', log_config_file)
+    if logger is None:
+        print("Failed to configure logging.\n")
+        sys.exit(-1)
+    return logger
+
+
+def create_replication_job_for_manager(s3_config, object_info):
+    job_dict = replication_job_template_for_manager()
+
+    # Update the fields in template.
+    job_dict["Object-Name"] = object_info["object_name"]
+    job_dict["Object-URI"] = s3_config.source_bucket_name + \
+        '\\\\' + object_info["object_name"]
+    job_dict["System-Defined"]["Content-Length"] = object_info["size"]
+    job_dict["System-Defined"]["Content-MD5"] = object_info["md5"]
+
+    job_dict["User-Defined"]["x-amz-meta-target-site"] = \
+        s3_config.s3_service_name
+
```

Comment:
update target bucket name

Reply:
This worked as I used default target name for my test that's why this went unrecognized. Glad that you caught this now itself.